lucene/core/src/java/org/apache/lucene/index/FieldInfos.java

```diff
@@ -63,21 +62,30 @@ public FieldInfos(FieldInfo[] infos) {
     boolean hasNorms = false;
     boolean hasDocValues = false;
     boolean hasPointValues = false;
-    
-    TreeMap<Integer, FieldInfo> byNumber = new TreeMap<>();
+
+    int size = 0; // number of elements in byNumberTemp
+    int capacity = 10; // byNumberTemp's capacity
+    FieldInfo[] byNumberTemp = new FieldInfo[capacity];
```

Comment:
what not using `ArrayList` ?

Reply:
as far as I'm concerned, I'm fine with either option: array or ArrayList.

Reply:
@diegoceccarelli thanks for a suggestion. I considered `ArrayList` as well, but the code was even more complex. `byNumberTemp` could potentially be a sparse array, where some positions don't have values.  You also need to `ensureCapacity` of the ArrayList, and `add` null elements.

Reply:
That's true :)  what about having an `ArrayList` and then a separate private method `ensureSize` ?

```
private static void ensureSize(ArrayList<?> list, int size) {
    list.ensureCapacity(size);
    while (list.size() < size) {
        list.add(null);
    }
}

```

In this way you will not have to care about the logic behind the resizing and you will have less variables around.

Reply:
This looks more complex to me than manipulating the array directly?